Source/WebCore/animation/WebAnimation.cpp

```diff
@@ -1347,11 +1347,20 @@ void WebAnimation::tick()
     updateFinishedState(DidSeek::No, SynchronouslyNotify::Yes);
     m_shouldSkipUpdatingFinishedStateWhenResolving = true;
 
-    // Run pending tasks, if any.
-    if (hasPendingPauseTask())
-        runPendingPauseTask();
-    if (hasPendingPlayTask())
-        runPendingPlayTask();
+    auto runPendingTasks = [&]() {
+        if (!pending())
+            return;
+
+        if (m_effect && m_effect->preventsAnimationReadiness())
+            return;
+
+        if (hasPendingPauseTask())
+            runPendingPauseTask();
+        if (hasPendingPlayTask())
+            runPendingPlayTask();
+    };
+
+    runPendingTasks();
```

Comment:
I don't see the point of putting this in a block/function.

Reply:
I thought it was better to write it this way with conditions that are true and return than wrap the existing calls in a `if (!m_effect || !m_effect->preventsAnimationReadiness())` clause, but I'll make that change.